pkgs/applications/audio/furnace/default.nix

```diff
@@ -0,0 +1,89 @@
+{ stdenv
+, lib
+, nix-update-script
+, fetchFromGitHub
+, fetchpatch
+, cmake
+, pkg-config
+, makeWrapper
+, fmt_8
+, libsndfile
+, SDL2
+, zlib
+, withJACK ? stdenv.hostPlatform.isUnix
+, libjack2
+, withGUI ? true
+}:
+
+let
+  version = "0.5.4";
+  src = fetchFromGitHub {
+    owner = "tildearrow";
+    repo = "furnace";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-XAdawmcu3bUBxwzDeRYd/hkao2FXsnUDfXEGk4jEG+A=";
+  };
+  # SDL 2.0.18+ is needed, not packages yet
+  # https://github.com/NixOS/nixpkgs/issues/154137
+  vendoredSDL2 = SDL2.overrideAttrs (oa: {
+    version = "2.0.x-furnace";
+    src = "${src}/extern/SDL";
+    patches = [];
+  });
+in
+stdenv.mkDerivation rec {
+  pname = "furnace";
+
+  inherit version src;
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ] ++ lib.optional stdenv.hostPlatform.isDarwin makeWrapper;
+
+  buildInputs = [
+    fmt_8
+    libsndfile
+    vendoredSDL2
+    zlib
+  ] ++ lib.optional withJACK libjack2;
+
+  cmakeFlags = [
+    "-DBUILD_GUI=${if withGUI then "ON" else "OFF"}"
+    "-DSYSTEM_FMT=ON"
+    "-DSYSTEM_LIBSNDFILE=ON"
+    "-DSYSTEM_ZLIB=ON"
+    "-DSYSTEM_SDL2=ON"
+    "-DWITH_JACK=${if withJACK then "ON" else "OFF"}"
+    "-DWARNINGS_ARE_ERRORS=ON"
+  ];
+
+  postInstall = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    # Normal CMake install phase on Darwin only installs the binary, the user is expected to use CPack to build a
+    # bundle. That adds alot of overhead for not much benefit (CPack is currently abit broken, and needs impure access
+    # to /usr/bin/hdiutil). So we'll manually assemble & install everything instead.
+
+    mkdir -p $out/{Applications/Furnace.app/Contents/{MacOS,Resources},share/{,doc,licenses}/furnace}
+    mv $out/{bin,Applications/Furnace.app/Contents/MacOS}/furnace
+    makeWrapper $out/{Applications/Furnace.app/Contents/MacOS,bin}/furnace
+
+    install -m644 {../res,$out/Applications/Furnace.app/Contents}/Info.plist
+    install -m644 ../res/icon.icns $out/Applications/Furnace.app/Contents/Resources/Furnace.icns
+    install -m644 {..,$out/share/licenses/furnace}/LICENSE
+    cp -r ../papers $out/share/doc/furnace/
+    cp -r ../demos $out/share/furnace/
+  '';
+
+  passthru.updateScript = nix-update-script {
+    attrPath = pname;
+  };
+
+  meta = with lib; {
+    description = "Multi-system chiptune tracker compatible with DefleMask modules";
+    inherit (src.meta) homepage;
+    license = with licenses; [ gpl2Plus ];
```

Comment:
Where did you find the "Plus" part? I couldn't find the "or any later version" wording anywhere in the repo

Reply:
I asked the developer about it.

![Bildschirmfoto von 2022-02-07 16-58-39](https://user-images.githubusercontent.com/23431373/152824392-d7158dbc-9b31-4142-ad22-f2bb40ea0716.png)

![Bildschirmfoto von 2022-02-07 17-00-25](https://user-images.githubusercontent.com/23431373/152824532-622582ec-02c9-4792-988f-b68e7c25b877.png)